lib/tarfiles.py

```diff
@@ -510,8 +507,11 @@ def publish(self, tarfile: str) -> requests.Response:
         if self.verbose:
             print(f"Calling URL {url}")
         if self.token:
-            return requests.post(url, auth=TokenAuth(self.token), data=tarfile)
-        return requests.post(url, cert=(self.proxy, self.proxy), data=tarfile)
+            with open(tarfilename, "rb") as tarfile:
+                return requests.post(url, auth=TokenAuth(self.token), data=tarfile)
+        res = requests.post(url, cert=(self.proxy, self.proxy), data=tarfile)
```

Comment:
If we don't have a token, then I believe this and the next line will fail with a `NameError` since `tarfile` isn't defined in the local scope.   Unless github is messing with the view, I believe the intended scoping (no `close()` necessary) is:

```python
with open(tarfilename, "rb") as tarfile:
    if self.token:
        return requests.post(url, auth=TokenAuth(self.token), data=tarfile)  # Token
    return requests.post(url, cert=(self.proxy, self.proxy), data=tarfile)  # Proxy
```

Is that right?

Reply:
Yes.   Fixed. Of course, if we can't get  a token, we won't get this far... but better to have it right.

Reply:
Thanks!